File: src/main/java/com/smpcore/sidebar/SidebarListener.java
package com.smpcore.sidebar;

import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

public final class SidebarListener implements Listener {
    private final SidebarManager sidebarManager;

    public SidebarListener(SidebarManager sidebarManager) {
        this.sidebarManager = sidebarManager;
    }

    @EventHandler
    public void onJoin(PlayerJoinEvent event) {
        sidebarManager.onJoin(event.getPlayer());
    }

    @EventHandler
    public void onQuit(PlayerQuitEvent event) {
        sidebarManager.onQuit(event.getPlayer());
    }
}

File: src/main/kotlin/com/smpcore/sidebar/SidebarListener.kt
package com.smpcore.sidebar

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

class SidebarListener(private val sidebarManager: SidebarManager) : Listener {
    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        sidebarManager.onJoin(event.player)
    }

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        sidebarManager.onQuit(event.player)
    }
}
